### PR TITLE
update do_svd

### PR DESCRIPTION
**Pull request by double-y**
### Description
* resolve https://github.com/exploratory-io/tam/issues/4739
* resolve https://github.com/exploratory-io/tam/issues/5856

### Checklist

Make sure you have performed following items before submitting this pull request.
If not, please describe the reason.  

- [x] Add test cases for this fix/enhancement
- [ ] Pass devtools::check()
- [x] Pass devtools::test()
- [ ] Test installing from github
- [ ] Tested with Exploratory

### CODE REVIEW

File: R/do_svd.R [hunk 1]
```diff
@@ -0,0 +1,308 @@
...
+      dplyr::select_(.dots=select_dots) %>%
+      as.matrix() %>%
+      na.omit()
+
+    if(centering){
+      # move the origin to center of data
+      matrix <- sweep(matrix, 2, colMeans(matrix), "-")
+    }
+    if(type=="group"){
+      # u matrix in svd can be regarded as coordinations of groups in reduced dimension
```
Comment (hideaki) [inline R/do_svd.R]: took some time to realize result$u is u matrix.
comment like, "u matrix (result$u) in svd can..." would help.

File: R/do_svd.R [hunk 2]
```diff
@@ -0,0 +1,308 @@
...
+
+  value_colname <- avoid_conflict(grouped_col, "value")
+
+  # this is executed on each group
+  do_svd_each <- function(df){
+    # create matrix by selected columns
+    matrix <- df %>%
+      dplyr::select_(.dots=select_dots) %>%
+      as.matrix() %>%
+      na.omit()
```
Comment (hideaki) [inline R/do_svd.R]: Comment on na.omit()'s behavior (i.e. removing row) would help.

File: R/do_svd.R [hunk 3]
```diff
@@ -0,0 +1,308 @@
...
+  ret <- df %>%
+    dplyr::do_(.dots=setNames(list(~do_svd_each(.)), tmp_col)) %>%
+    unnest_with_drop_(tmp_col)
+
+  ret
+}
+
+#' Calculate distance of each pair of groups.
+#' @param df data frame in tidy format
+#' @param group A column you want to calculate the correlations for.
```
Comment (hideaki) [inline R/do_svd.R]: The parameters comment does not seem to match with actual parameters. For example, I don't see type here.

File: R/do_svd.R [hunk 4]
```diff
@@ -0,0 +1,322 @@
...
+      # Move the origin to center of data
+      # (Mean of each column)
+      #
+      # Explanation from http://genomicsclass.github.io/book/pages/pca_svd.html
+      # "The second argument specifies we want to operate on the columns
+      # (1 would be used for rows), and the third and fourth arguments
+      # specify that we want to subtract the column means."
+      matrix <- sweep(matrix, 2, colMeans(matrix), "-")
+    }
+    if(type=="group"){
```
Comment (hideaki) [inline R/do_svd.R]: comment on what is done in this if statement as a whole would help.

File: R/do_svd.R [hunk 5]
```diff
@@ -0,0 +1,322 @@
...
+      # u matrix in svd can be regarded as coordinations of groups in reduced dimension
+      result <- svd(matrix, nu=n_component, nv=0)
+      mat <- result$u
+
+      if (output=="wide") {
```
Comment (hideaki) [inline R/do_svd.R]: comment on what is done in this if statement as a whole would help.

File: R/do_svd.R (hunk 2 above)
Comment (double-y) [inline R/do_svd.R]: Ok, will add a comment

File: R/do_svd.R (hunk 3 above)
Comment (double-y) [inline R/do_svd.R]: Will update them

File: R/do_svd.R (hunk 4 above)
Comment (double-y) [inline R/do_svd.R]: Ok, will add it

File: R/do_svd.R (hunk 5 above)
Comment (double-y) [inline R/do_svd.R]: Ok, will add it

**Pull request merged**